src/transformers/models/cohere/modeling_cohere.py

```diff
@@ -1003,9 +987,9 @@ def _update_causal_mask(
         dtype, device = input_tensor.dtype, input_tensor.device
         min_dtype = torch.finfo(dtype).min
         sequence_length = input_tensor.shape[1]
-        if hasattr(getattr(self.layers[0], "self_attn", {}), "past_key_value"):  # static cache
-            target_length = self.config.max_position_embeddings
-        else:  # dynamic cache
+        if using_static_cache:
+            target_length = past_key_values.get_max_length()
```

Comment:
can't we always use get_max_length()?

Reply:
`get_max_length()` is `None` in the dynamic caches

Reply:
It should be `seq_length`

Reply:
but alright